Lesson7

GareevaFarida opened the pull request:
Выполнена домашняя работа по уроку 7.
1. Реализован обмен приватными сообщениями между пользователями. Для выбора собеседника нужно ввести имя в текствое поле. Если собеседник не в сети, пользователь получает сообщение от сервера.
2. В текст сообщения добавлена дата.
3. Все подключенные пользователи получают сообщение от сервера о подключении или выходе другого пользователя.

GareevaFarida committed:
Домашняя работа по уроку 6.
Не поняла, как можно удалить из списка потов вышедших из чата. Сервер рассылает сообщения всем когда-либо входившим в чат.

GareevaFarida committed:
Домашняя работа по уроку 7

GareevaFarida committed:
Домашняя работа по уроку 7

usharik commented on src/lesson6/ChatServer.java:
```diff
@@ -0,0 +1,114 @@
+            return;
+        }
+        try {
+            handler.getOut().writeUTF(msg);
```
Так тоже можно, но лучше было бы этот код перенести в обработчик клиента.

GareevaFarida committed:
Домашняя работа по уроку 8.
В диалоге пользователя появился список пользователей в online/
Запущен поток ожидания авторизации.
Константы вынесены в отдельный файл.
